code/modules/integrated_electronics/subtypes/arithmetic.dm

```diff
@@ -311,3 +311,34 @@
 	set_pin_data(IC_OUTPUT, 1, result)
 	push_data()
 	activate_pin(2)
+
+// -Max- //
+/obj/item/integrated_circuit/arithmetic/max
+	name = "max circuit"
+	desc = "This circuit sends out the highest number."
+	extended_desc = "The highest number is put out. Null is ignored."
+	icon_state = "addition"
+	var/state=0
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/arithmetic/max/do_work()
+	var/result
+	for(var/k in 1 to inputs.len)
+		var/I = get_pin_data(IC_INPUT, k)
+		if(isnum(I))
+			if(!(isnum(result)))
+				result = I
+			else
+				if(I > result && state == 0) result = I
+				else if(I < result && state == 1) result = I
```

Comment:
The above 5 lines can be simplified to

```dm
if(result == null || (state && I < result) || (!state && I > result))
    result = I
```
Never put control flow body on the same line outside of macros please